Argo/Globals/Parser.swift

```diff
@@ -0,0 +1,22 @@
+import Box
+
+public enum Parser<T> {
+  case Success(Box<T>)
+  case TypeMismatch(String)
+  case MissingKey(String)
+
+  public var value: T? {
+    switch self {
+    case let .Success(box): return box.value
+    default: return .None
+    }
+  }
+}
+
+func optional<A>(p: Parser<A>) -> Parser<A?> {
```

Comment:
Sorry, I do think this should be a static function on `Parser` to avoid polluting the global scope.